lib/stm32/common/usart_common_all.c

```diff
@@ -52,17 +52,32 @@ usart_reg_base
 
 void usart_set_baudrate(uint32_t usart, uint32_t baud)
 {
+
+#if defined STM32H7
```

Comment:
you've opened the box here :)  This is actually something that's an option (depending on the peripheral) on f4, f7, l4, l4+, g4 and I think even g0.   So.... yes, I see what the problem you're trying to solve is, but we might need to go back further and work out something that's going to be a bit more maintainable

Reply:
Yes, this happens on every platform, and the existing model was to ifdef out the special cases. I followed this, and my implementation doesn't block out the other platforms, it simply bypasses all of the other if/else logic for the H7 as it gets its clock in a differenter way.

Happy to refactor or have this refactored, but seems like it could/should be done outside of the scope of this review.

Reply:
it can be out of scope, but then I'd like to remove this stuff altogether, or limit it to "it only works if you use ahb clocks for usart, as done on "legacy" platforms"  I don want to have this come into the usart code as part of "just supporting some pll options on h7"

Reply:
Okay, I see...I will remove this alltogether. 

As a suggestion, I would maybe just add an entry point that is `usart_set_baudrate_with_clock(uint32_t usart, uint32_t clk, uint32_t baud)` which can then be called by custom functions for each platform rather than having all of the ifdefing?  Just a thought.  I'm simply going to remove these changes.

Reply:
Actually...I can't simply leave as-is unfortunately. I will do some refactoring and you can let me know if you are okay with it...

Reply:
Alright...after making various attempts at refactoring, the best way I could determine to do this is to move the H7 implementation out of the file and have the default implementation conditionally compiled. This achieves your goal of not mixing this more complicated clocking scheme into the "legacy" implementation.  Unfortunately, due to how everything is structured, there was not an easy way to move this function out of this file without potential breaking users. Possibly we could create a few variant `usart_xxx_setbaud.c` files and update each legacy type to build in the correct one, but this seemed a bit more risky than what I've implemented. Let me know what you think.